systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -2128,6 +2137,344 @@ void testKafkaOffsetsReplicationFactorHigherThanReplicas() {
+
+        kafkaConfigurationFromPod = cmdKubeClient().execInPod(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "/bin/bash", "-c", "bin/kafka-configs.sh --bootstrap-server localhost:9092 --entity-type brokers --entity-name 0 --describe").out();
+        assertThat(kafkaConfigurationFromPod, containsString("unclean.leader.election.enable=true"));
+    }
+
+    @Test
+    @Tag(EXTERNAL_CLIENTS_USED)
+    void testDynamicConfigurationExternalTls() throws IOException, InterruptedException, ExecutionException, TimeoutException {
+        int kafkaReplicas = 2;
+        Map<String, Object> kafkaConfig = new HashMap<>();
+        kafkaConfig.put("offsets.topic.replication.factor", "1");
+        kafkaConfig.put("transaction.state.log.replication.factor", "1");
+        kafkaConfig.put("default.replication.factor", "1");
+        kafkaConfig.put("log.message.format.version", "2.4");
+
+        BasicExternalKafkaClient basicExternalKafkaClientTls = new BasicExternalKafkaClient.Builder()
+                .withTopicName(TOPIC_NAME)
+                .withNamespaceName(NAMESPACE)
+                .withClusterName(CLUSTER_NAME)
+                .withMessageCount(MESSAGE_COUNT)
+                .withConsumerGroupName(CONSUMER_GROUP_NAME + "-" + rng.nextInt(Integer.MAX_VALUE))
+                .withSecurityProtocol(SecurityProtocol.SSL)
+                .build();
+
+        BasicExternalKafkaClient basicExternalKafkaClient = new BasicExternalKafkaClient.Builder()
+                .withTopicName(TOPIC_NAME)
+                .withNamespaceName(NAMESPACE)
+                .withClusterName(CLUSTER_NAME)
+                .withMessageCount(MESSAGE_COUNT)
+                .withConsumerGroupName(CONSUMER_GROUP_NAME + "-" + rng.nextInt(Integer.MAX_VALUE))
+                .withSecurityProtocol(SecurityProtocol.PLAINTEXT)
+                .build();
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, kafkaReplicas, 1)
+                .editSpec()
+                    .editKafka()
+                        .withNewListeners()
+                            .withNewKafkaListenerExternalNodePort()
+                                //.withTls(false)
+                            .endKafkaListenerExternalNodePort()
+                            .withNewPlain()
+                            .endPlain()
+                        .endListeners()
+                        .withConfig(kafkaConfig)
+                    .endKafka()
+                .endSpec()
+                .done();
+
+        KafkaTopicResource.topic(CLUSTER_NAME, TOPIC_NAME).done();
+
+        String userName = "alice";
+        KafkaUserResource.tlsUser(CLUSTER_NAME, userName).done();
+        waitFor("Wait for secrets became available", Constants.GLOBAL_POLL_INTERVAL, Constants.TIMEOUT_FOR_GET_SECRETS,
+            () -> kubeClient().getSecret(userName) != null,
+            () -> LOGGER.error("Couldn't find user secret {}", kubeClient().listSecrets()));
```

Comment:
There is no need for this waiting...If you take a look on the implementation in the `tlsUser()` method you will find 
```
private static KafkaUser waitFor(KafkaUser kafkaUser) {
        LOGGER.info("Waiting for Kafka User {}", kafkaUser.getMetadata().getName());
        SecretUtils.waitForSecretReady(kafkaUser.getMetadata().getName());
        KafkaUserUtils.waitForKafkaUserCreation(kafkaUser.getMetadata().getName());
        LOGGER.info("Kafka User {} is ready", kafkaUser.getMetadata().getName());
        return kafkaUser;
    }
```
It's already waiting on the secret in the 2nd and 3rd line.

Reply:
It is copypaste error :(